hedera-node/src/main/java/com/hedera/services/txns/contract/operation/HederaCreateOperation.java

```diff
@@ -47,24 +48,33 @@ public Gas cost(final MessageFrame frame) {
 
 		//TODO charge for memory expansion cost?
 		//final Gas memoryGasCost = gasCalculator().memoryExpansionGasCost(frame, initCodeOffset, initCodeLength);
-		HashMap<Address, Long> map = frame.getMessageFrameStack().getLast().getContextVariable("expiries");
+
 		long expiry = 0;
```

Comment:
I am wondering whether it would be better to use directly `long` as a context variable and get it here. 
In ContractCreateTransitionLogic it will be populated and for the rest of the tx it will be `0` passed in the `super.execute`.

And here we can assign the expiry to the context variable (by default) and then iterate through the message frames. That way we will skip the following code:
```Java
/* if this is the initial frame from the deque, check context vars first */
			if (!framesIterator.hasNext()) {
				Optional<Long> expiryOptional = messageFrame.getContextVariable("expiry");
				if (expiryOptional.isPresent()) {
					expiry = expiryOptional.get();
					break;
				}
			}
```

This would  ease the cognitive load

Reply:
Currently, since the logic of retrieving the value from the context variables is in the while loop, we know we're retrieving it only when the last frame is reached and we break from the while if it exists in the context vars, meaning that even if the sender account exists and has expiry, we would use the value from the context vars. If we change it to retrieve the value from context first and then iterate over the frames, we would use the value from the senderaccount, unless we add a check like (!framesIterator.hasNext && priority != 0) which is not much better than the current logic imho